goth/assertions/monitor.py

```diff
@@ -20,45 +44,64 @@ class EventMonitor(Generic[E]):
     registered event.
     """
 
-    assertions: List[Assertion[E]]
-    """List of all assertions, active or finished."""
+    assertions: List[Tuple[Assertion[E], LogLevel]]
```

Comment:
I think we could use a dict here instead (i.e. `Dict[Assertion[E], LogLevel]`). That way, when we need to iterate over just the assertions we can use the collection of keys. Also, in the (unlikely) event of adding more attributes related to an assertion besides the log level, we can make the values store objects of some configuration class.

Reply:
I wasn't sure if we won't ever need the assertions to be ordered, so for example "more important" assertions are checked and reported before the less important ones. This is perhaps a bit contrived, anyway: I'll make it an `OrderedDict` to have our cake and eat it.